### ISSUE TITLE
Fix failing stats tests for the new PostgreSQL backend

### ISSUE THREAD

**Issue by rumyantseva** [FerretDB/FerretDB#3521]
### What should be done?

Figure out why tests fail and provide the necessary fixes.

### Where?

```
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestCommandsAdministrationCollStats'
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestCommandsAdministrationCollStatsWithScale'
task test-integration-postgresql SHARD_INDEX=0 SHARD_TOTAL=0 TEST_RUN='TestCommandsAdministrationDataSize'
```

See also this issue mentioned in the code - https://github.com/FerretDB/FerretDB/pull/3522/files.

### Definition of Done

- fixes provided;
- integration/compatibility tests updated.

**Comment by chilagrow** [comment]
@rumyantseva Do you any CI or error on how it fails?

For me, it always passes locally. After a couple of attempt, I got CI to fail like this. It indicates that any call to Stats is slow because we call `ANALYZE`. https://github.com/FerretDB/FerretDB/actions/runs/6465143749/job/17550835758?pr=3538

**Comment by rumyantseva** [comment]
Yeah, messages like `Max difference between 200 and 463 allowed is 200, but difference was -263` (from your link) is how sometimes (but not always) `TestCommandsAdministrationDataSize` fails on my machine. I don't really agree that it's related to the fact that any call to Stats is slow, it's rather we either need to change the delta in the test, or the code.

The other two tests `TestCommandsAdministrationCollStats` and `TestCommandsAdministrationCollStatsWithScale` don't fail anymore. But it might be possible that the first time I was running them had been before #3478 was merged, so maybe #3478 actually is the PR that fixed them.

I suggest we just remove `Skip` from `TestCommandsAdministrationCollStats` and `TestCommandsAdministrationCollStatsWithScale` and figure out the right deltas for `TestCommandsAdministrationDataSize`.

**Comment by chilagrow** [comment]
> @chilagrow yeah, messages like `Max difference between 200 and 463 allowed is 200, but difference was -263` (from your link) is how sometimes (but not always) `TestCommandsAdministrationDataSize` fails on my machine. I don't really agree that it's related to the fact that any call to Stats is slow, it's rather we either need to change the delta in the test, or the code.
> 
> The other two tests `TestCommandsAdministrationCollStats` and `TestCommandsAdministrationCollStatsWithScale` don't fail anymore. But it might be possible that the first time I was running them had been before #3478 was merged, so maybe #3478 actually is the PR that fixed them.
> 
> I suggest we just remove `Skip` from `TestCommandsAdministrationCollStats` and `TestCommandsAdministrationCollStatsWithScale` and figure out the right deltas for `TestCommandsAdministrationDataSize`.

@rumyantseva, sure I'll try to figure out delta for dataSize test. As for CollStats tests, I'll remove skip :hugs:
